### PR TITLE
Date: add relative time translations for moment.js

### LINKED ISSUE #53888: Site editor: last modified time is not translatable
### Description

Some strings from global styles revisions are not translatable.

### Step-by-step reproduction instructions

- Switch your WordPress environment to a language other than English that is 100% translated (e.g.: Spanish).
- Visit "Appearance > Styles".
- Provided you have some style revision you'll see the last modified time at the bottom. Note how the time specifier (minutes, hours, days) is not translated:

Captura de ecrã 2023-08-23, às 12 02 55

- Click on it, which opens the global styles revisions sidebar. There's the same issue there (note the "minutes", "a day", or "days" strings):

Captura de ecrã 2023-08-23, às 12 03 16

### Environment info

I tested this on WordPress 6.3, though the string comes from Gutenberg.

The corresponding lines point to the use of `humanTimeDiff` function [here](https://github.com/WordPress/gutenberg/blob/update/usage-of-global-settings/packages/edit-site/src/components/sidebar-navigation-screen-details-footer/index.js#L32) and [here](https://github.com/WordPress/gutenberg/blob/update/usage-of-global-settings/packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js#L134).

### PR DESCRIPTION
**Pull request by ramonjd**
Resolves https://github.com/WordPress/gutenberg/issues/53888

## What?
Overwrites `wp-date` script string to include new `relativeTime` translations.

The inline script is generated in Core here: https://github.com/WordPress/wordpress-develop/blob/bf00a673a5f1d7f4707a6f914e1408526a8dc06a/src/wp-includes/script-loader.php#L430 and will need to be synced.

See: https://momentjscom.readthedocs.io/en/latest/moment/07-customization/07-relative-time/

## Why?
The [settings strings](https://github.com/WordPress/gutenberg/blob/eed863a1aacecf166b43eaa90077ccd0f287ad41/packages/date/src/index.js#L111-L111) were never synced to Core. 

Now that we're using `relativeTime` in [humanTimeDiff](https://github.com/WordPress/gutenberg/blob/eed863a1aacecf166b43eaa90077ccd0f287ad41/packages/date/src/index.js#L592C8-L592C44), we need the translations.

Unfortunately, `%d minutes` and others have not yet been translated.

## How?

Hooking into `wp_default_scripts` and overwriting the settings of the registered `wp-date` script.

The other option is to re-add the inline script, which will also work, but it will output 2 x inline scripts, e.g., 

```html
<script id="wp-date-js-after">
wp.date.setSettings( ...oldSettings );
wp.date.setSettings( ...newSettings );
</script>
```

## Testing Instructions
Change your editor/site to another language via Settings > General
Since the relevant translations have not been added, you can check that the `wp.date` settings have been updated by heading to `/wp-admin/site-editor.php` and, in the browser console, running `wp.date.getSettings()`

The `relative` object should match the requirements of https://momentjscom.readthedocs.io/en/latest/moment/07-customization/07-relative-time/

```js
{
    "l10n": {
        "locale": "de_DE",
        "months": [
            "Januar",
            "Februar",
            "März",
            "April",
            "Mai",
            "Juni",
            "Juli",
            "August",
            "September",
            "Oktober",
            "November",
            "Dezember"
        ],
        "monthsShort": [
            "Jan",
            "Feb",
            "Mrz",
            "Apr",
            "Mai",
            "Jun",
            "Jul",
            "Aug",
            "Sep",
            "Okt",
            "Nov",
            "Dez"
        ],
        "weekdays": [
            "Sonntag",
            "Montag",
            "Dienstag",
            "Mittwoch",
            "Donnerstag",
            "Freitag",
            "Samstag"
        ],
        "weekdaysShort": [
            "So",
            "Mo",
            "Di",
            "Mi",
            "Do",
            "Fr",
            "Sa"
        ],
        "meridiem": {
            "am": "am",
            "pm": "pm",
            "AM": "AM",
            "PM": "PM"
        },
        "relative": {
            "future": "%s ab jetzt",
            "past": "vor %s",
            "s": "a second",
            "ss": "%d seconds",
            "m": "a minute",
            "mm": "%d minutes",
            "h": "an hour",
            "hh": "%d hours",
            "d": "a day",
            "dd": "%d days",
            "M": "a month",
            "MM": "%d months",
            "y": "a year",
            "yy": "%d years"
        },
        "startOfWeek": 1
    },
    "formats": {
        "time": "g:i a",
        "date": "F j, Y",
        "datetime": "j. F Y G:i",
        "datetimeAbbreviated": "j. M. Y G:i"
    },
    "timezone": {
        "offset": 0,
        "string": "",
        "abbr": ""
    }
}
```

### CODE REVIEW

**Review by andrewserong — APPROVED** [review]
Thanks so much for the fix @ramonjd!

✅ Confirmed the missing strings prior to applying this PR
✅ With this PR applied, the settings are identical as without it, except for the additional relative strings (when running `JSON.stringify( wp.date.getSettings() );` to compare strings)
✅ Visually, the code here otherwise matches what's in core

LGTM! ✨

File: lib/client-assets.php [hunk 1]
```diff
@@ -594,3 +594,96 @@ function gutenberg_register_vendor_scripts( $scripts ) {
...
+							'relative'      => array(
+								/* translators: %s: Duration. */
+								'future' => __( '%s from now', 'default' ),
+								/* translators: %s: Duration. */
+								'past'   => __( '%s ago', 'default' ),
+								/* translators: One second from or to a particular datetime, e.g., "a second ago" or "a second from now". */
+								's'      => __( 'a second', 'default' ),
+								/* translators: %s: Duration in seconds from or to a particular datetime, e.g., "4 seconds ago" or "4 seconds from now". */
+								'ss'     => __( '%d seconds', 'default' ),
+								/* translators: One minute from or to a particular datetime, e.g., "4 seconds ago" or "in 4 seconds from now". */
```
Comment (t-hamano) [inline lib/client-assets.php]:
```suggestion
								/* translators: One minute from or to a particular datetime, e.g., "a minute ago" or "a minute from now". */
```

**Review by t-hamano — APPROVED** [review]
LGTM!

My concern is that this code might be better written in `lib/compat/wordpress-6.4/script-loader.php`.

I think this hook expects all relative times to translate correctly on all WordPress versions supported by the Gutenberg plugin. And if the minimum WordPress version that the Gutenberg plugin supports has this code, this hook will not be needed.

I think the `lib/compat/wordpres-{X.Y}` directory will be removed as the Gutenberg plugin supports more WordPress versions, so you can avoid forgetting to remove this hook.

What do you think?

**Comment by ramonjd** [comment]
> I think this hook expects all relative times to translate correctly on all WordPress versions supported by the Gutenberg plugin. And if the minimum WordPress version that the Gutenberg plugin supports has this code, this hook will not be needed.

Yes, it totally makes more sense to store this under a `compat` folder. I'll move it there 👍🏻

**Pull request merged**